pkg/pillar/cmd/zedkube/kubeinterface.go

```diff
@@ -78,6 +83,116 @@ func runKubeConfig(ctx *zedkubeContext, config, oldconfig *types.EdgeNodeCluster
 	publishKubeConfigStatus(ctx, config)
 }
 
+type Drainer struct {
+	Client              *kubernetes.Clientset
+	Force               bool
+	IgnoreAllDaemonSets bool
+	DeleteEmptyDirData  bool
+	GracePeriodSeconds  int64
+	Timeout             time.Duration
+	Out                 *os.File
+	ErrOut              *os.File
+}
+
+func RunNodeDrain(drainer *Drainer, nodeName string) error {
+	pods, err := drainer.Client.CoreV1().Pods("").List(context.Background(), metav1.ListOptions{
+		FieldSelector: "spec.nodeName=" + nodeName,
+	})
```

Comment:
Can we open this up to allow a pod selector string allowing the kubernetes style label syntax (eg. "app!=some-pod-app-name")?  The upcoming upgrade path will make use of something like this.
```suggestion
type Drainer struct {
	Client              *kubernetes.Clientset
	Force               bool
	IgnoreAllDaemonSets bool
	DeleteEmptyDirData  bool
	GracePeriodSeconds  int64
	Timeout             time.Duration
	Out                 *os.File
	ErrOut              *os.File
	PodSelector    string
}

func RunNodeDrain(drainer *Drainer, nodeName string) error {
	pods, err := drainer.Client.CoreV1().Pods("").List(context.Background(), metav1.ListOptions{
		FieldSelector: "spec.nodeName=" + nodeName,
		LabelSelector: drainer.PodSelector
	})
```

Reply:
sounds good.